Employee/EmployeeManagement.java

```diff
@@ -0,0 +1,366 @@
+            System.out.println("1. Add an employee");
+            System.out.println("2. Delete an employee");
+            System.out.println("3. Search employees");
+            System.out.println("4. Display All employees");
+            System.out.println("5. Quit");
+            int choice;
+            try{
+            choice = scanner.nextInt();}
+            catch(InputMismatchException e){
+                System.out.println("Invalid choice");
+                return;
+            }
+            scanner.nextLine();
+            switch (choice) {
+                case 1:
+                    addEmployee(scanner);
+                    break;
+                case 2:
+                    deleteEmployee(scanner);
+                    break;
+                case 3:
+                    searchEmployees(scanner);
+                    break;
+                case 4:
+                    displayAllEmployees();
+                 break;
+                case 5:
+                    quit = true;
+                    break;
+                default:
+                    System.out.println("Invalid choice");
+                    break;
+            }
+        }
+    }
+
+    private static List<Employee> loadEmployeesFromFile() {
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+
+        List<Employee> employees = new ArrayList<>();
+        try (Scanner scanner = new Scanner(new File(FILE_NAME))) {
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                String[] parts = line.split(",");
+                if (parts.length == 5) {
+                    Employee employee = new Employee();
+                    employee.setId(Integer.parseInt(parts[0]));
+                    employee.setName(parts[1]);
+
+                    
+                    employee.setEmail(parts[2]);    
+                    
+                    employee.setAge(Integer.parseInt(parts[3]));
+                    try {
+                        employee.setDob(dateFormat.parse(parts[4]));
+                    } catch (ParseException e) {
+                        e.printStackTrace();
+                        System.out.println( "Invalid Date");
+                    }
+                    employees.add(employee);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            // If the file is not found, just return an empty list
+        }
+        return employees;
+    }
+
+    private static void saveEmployeesToFile(List<Employee> employees) {
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+
+        try (PrintWriter writer = new PrintWriter(new FileWriter(FILE_NAME))) {
+            for (Employee employee : employees) {
+                writer.println(employee.getId() + "," + employee.getName() + "," + employee.getEmail() + "," + employee.getAge() + "," + dateFormat.format(employee.getDob()));
```

Comment:
Is there any method to append the new employee to the file because rewriting all the employees may be a costly operation if the number of existing employees is large? Please check.